src/vm/methodtable.cpp

```diff
@@ -2703,17 +2722,33 @@ bool MethodTable::ClassifyEightBytesWithNativeLayout(SystemVStructRegisterPassin
             {
                 MethodTable* pFieldMT = ((FieldMarshaler_FixedArray*)pFieldMarshaler)->GetElementMethodTable();
 
-                bool inEmbeddedStructPrev = helperPtr->inEmbeddedStruct;
-                helperPtr->inEmbeddedStruct = true;
-                bool structRet = pFieldMT->ClassifyEightBytesWithNativeLayout(helperPtr, nestingLevel + 1, normalizedFieldOffset, useNativeLayout);
-                helperPtr->inEmbeddedStruct = inEmbeddedStructPrev;
+                unsigned int numElements = ((FieldMarshaler_FixedArray*)pFieldMarshaler)->GetNumElements();
+
+                bool structRet = true;
+
+                for (unsigned int i = 0; i < numElements; i++, normalizedFieldOffset += pFieldMT->GetNativeSize())
+                {
+                    bool inEmbeddedStructPrev = helperPtr->inEmbeddedStruct;
+                    helperPtr->inEmbeddedStruct = true;
+                    structRet = pFieldMT->ClassifyEightBytesWithNativeLayout(
```

Comment:
We only care about the last setter of `structRet`? Seems like at the very least we should assert something consistent about them, or is there something I am missing about this loop? (i.e. we only ever care about the last element?)

Reply:
The return value of `ClassifyEightBytesWithNativeLayout` only depends on the `this` pointer. Which makes sense since the return value says "this type can or cannot be enregistered based on its native layout".

The only code-paths where it will return something different based on the passed in parameters are paths where it asserts in debug mode because it breaks an invariant. So the value we assign to `structRet` would be the same every time.